echion/threads.h

```diff
@@ -113,32 +117,31 @@ void ThreadInfo::update_cpu_time()
 bool ThreadInfo::is_running()
 {
 #if defined PL_LINUX
-    char buffer[2048] = "";
-
-    std::ostringstream file_name_stream;
-    file_name_stream << "/proc/self/task/" << this->native_id << "/stat";
+    static char file_path[128];
 
-    int fd = open(file_name_stream.str().c_str(), O_RDONLY);
-    if (fd == -1)
-        return -1;
+    int path_len = snprintf(file_path, sizeof(file_path), "/proc/self/task/%ld/stat", this->native_id);
+    if (path_len < 0 || static_cast<size_t>(path_len) >= sizeof(buffer)) {
+        return false;  // Path construction failed or buffer too small
+    }
 
-    if (read(fd, buffer, 2047) == 0)
-    {
-        close(fd);
-        return -1;
+    // Open the file
+    int fd = open(file_path, O_RDONLY);
```

Comment:
Do you reckon we would be able to open this file once when we track the thread, and close it when we untrack it, and still get valid reads here?

Reply:
Yeah, I'd try that in a follow up PR. I'm thinking of having a mapping between native thread ids to file descriptors, and use `pread` to avoid frequent `open()` and `close()`. When `pread` fails, we can evict the entry from the map. Though not entirely sure whether this would work.

Reply:
Another thing to try would be using ebpf, where we can bundle up all proc fs operations into a single syscall, and return the results to echion.